src/code/z_lights.c

```diff
@@ -1,15 +1,83 @@
 #include "global.h"
 #include "assets/objects/gameplay_keep/gameplay_keep.h"
 
+// #define LIGHTS_DEBUG
+
+#if !(ENABLE_F3DEX3 || defined(F3DEX_GBI_PL))
+#error A microcode that supports point lighting is required
+#endif
```

Comment:
The F3DEX2 version in gamecube versions of OoT has point lighting support, so F3DEX2 with point lighting is fully supported with these engine changes. I dropped support for non-point-light F3DEX2 since it was never used in the first place and I didn't see much need for it, but I need to think about this again if we're worried about performance implications of always using ucode point lights.

Reply:
> I dropped support for non-point-light F3DEX2 since it was never used in the first place

What about when (eventually tm) decomp supports like oot 1.0, that wouldn't rock the f3dex2-PL ucode? What would be your plan then

Reply:
The plan would be don't use an outdated ucode version in any build and always use the current version

Reply:
Wait, you're saying that Master Quest Debug has been using F3DEX2 with point lighting this whole time???

Reply:
That's correct, they updated the ucode for gamecube and for some reason it was updated to a version with point light support.